src/common/cluster-types.ts

```diff
@@ -27,6 +29,37 @@ export type ClusterId = string;
  */
 export type UpdateClusterModel = Omit<ClusterModel, "id">;
 
+/**
+ * A type validator for `UpdateClusterModel` so that only expected types are present
+ */
+export const updateClusterModelChecker = Joi.object<UpdateClusterModel>({
+  kubeConfigPath: Joi.string()
+    .required()
+    .min(1),
+  contextName: Joi.string()
+    .required()
+    .min(1),
+  workspace: Joi.string()
+    .optional(),
```

Comment:
optional is default, right? why mark this one and not the others below?

Reply:
Fair point. I think we could probably remove this old and unused code too but I think that should be done in a separate PR.